examples_tests/41.VisibilityBuffer/main.cpp

```diff
@@ -3,815 +3,437 @@
-#extension GL_NV_fragment_shader_barycentric : enable
-#extension GL_AMD_shader_explicit_vertex_parameter : enable
+#define _NBL_VERT_INPUTS_DEFINED_
 
-#if !defined(IRR_GL_NV_fragment_shader_barycentric) && !defined(GL_AMD_shader_explicit_vertex_parameter) 
-    #error "GL_NV_fragment_shader_barycentric, nor GL_AMD_shader_explicit_vertex_parameter available on your GPU, or IrrlichtBaW messed up the support"
-#endif
+//#define nbl_glsl_VirtualAttribute_t uint
 
-#ifdef GL_AMD_shader_explicit_vertex_parameter
-    #error "Don't have AMD hardware to test https://gpuopen.com/stable-barycentric-coordinates/"
-#endif
+vec4 nbl_glsl_decodeRGB10A2_UNORM(in uint x)
+{
+	const uvec3 rgbMask = uvec3(0x3ffu);
+	const uvec4 shifted = uvec4(x,uvec3(x)>>uvec3(10,20,30));
+	return vec4(vec3(shifted.rgb&rgbMask),shifted.a)/vec4(vec3(rgbMask),3.0);
+}
 
-struct PerInstanceData
+vec4 nbl_glsl_decodeRGB10A2_SNORM(in uint x)
 {
-    mat4 modelViewProj;
-    vec3 Ka;
-    vec3 Kd;
-    vec3 Ks;
-    vec3 Ke;
-    uvec2 map_Ka_data;
-    uvec2 map_Kd_data;
-    uvec2 map_Ks_data;
-    uvec2 map_Ns_data;
-    uvec2 map_d_data;
-    uvec2 map_bump_data;
-    float Ns;
-    float d;
-    float Ni;
-    uint extra; //flags copied from MTL metadata
-};
+    uvec4 shifted = uvec4(x,uvec3(x)>>uvec3(10,20,30));
+    const uvec3 rgbMask = uvec3(0x3ffu);
+    const uvec3 rgbBias = uvec3(0x200u);
+    return max(vec4(vec3(shifted.rgb&rgbMask)-rgbBias,float(shifted.a)-2.0)/vec4(vec3(rgbBias-uvec3(1u)),1.0),vec4(-1.0));
+}
 
-layout(set=3,binding=0) uniform DrawCallBuffer {
-    uint InstanceIndex[];
-};
-layout(set=3,binding=1,row_major) readonly restrict buffer InstanceDataBuffer {
-    PerInstanceData instanceData[];
-};
-)";
+//pos
+layout(set = 0, binding = 0) uniform samplerBuffer MeshPackedData_R32G32B32_SFLOAT;
 
-constexpr const char* GLSL_VERT_OVERRIDE =
-R"(
-layout(location = 0) in vec3 vPos;
-#define _IRR_VERT_INPUTS_DEFINED_
+//uv
+layout(set = 0, binding = 1) uniform samplerBuffer MeshPackedData_R32G32_SFLOAT;
 
-#ifdef GL_AMD_shader_explicit_vertex_parameter
-    layout(location = 0) flat out vec4 flatPosition;
-    layout(location = 1)      out vec4 explicitPosition;
-#endif
-#define _IRR_VERT_OUTPUTS_DEFINED_
+//normal
+layout(set = 0, binding = 2) uniform usamplerBuffer MeshPackedData_A2B10G10R10_SNORM_PACK32;
 
-// no descriptors
-#define _IRR_VERT_SET0_BINDINGS_DEFINED_
-#define _IRR_VERT_SET1_BINDINGS_DEFINED_
-#define _IRR_VERT_SET2_BINDINGS_DEFINED_
-#define _IRR_VERT_SET3_BINDINGS_DEFINED_
+layout(set = 0, binding = 3) readonly buffer VertexDataOffsetTable
+{
+    int dataOffsetTable[];
+} vertexPosition;
 
-#include <nbl/builtin/glsl/vertex_utils/vertex_utils.glsl>
+layout(set = 0, binding = 4) readonly buffer VertexUVOffsetTable
+{
+    int dataOffsetTable[];
+} vertexUV;
 
-void main()
+layout(set = 0, binding = 5) readonly buffer VertexNormalOffsetTable
 {
-    uint instanceID = InstanceIndex[gl_DrawIndex];
+    int dataOffsetTable[];
+} vertexNormal;
```

Comment:
the virtual attribute handles to be packed into what was a Push Constant struct before (you should pack the push constant structs for all drawcalls into an unsized SSBO though, to enable the usage of MDI)